watertap/costing/units/gac.py

```diff
@@ -22,142 +22,99 @@
-        initialize=0,
-        units=pyo.units.USD_2020 * (pyo.units.m**3) ** -3,
-        doc="GAC concrete gravity contactor polynomial cost coefficient 3",
+        doc="number of off-line redundant GAC contactors in parallel",
+    )
+    blk.regen_frac = pyo.Var(
+        initialize=0.70,
+        units=pyo.units.dimensionless,
+        doc="fraction of spent GAC adsorbent that can be regenerated for reuse",
     )
+
+    # ---------------------------------------------------------------------
+    # correlation reference points
+
     blk.bed_mass_max_ref = pyo.Var(
         initialize=18143.7,
         units=pyo.units.kg,
-        doc="Reference maximum value of GAC mass needed for initial charge where "
+        doc="reference maximum value of GAC mass needed for initial charge where "
         "economy of scale no longer discounts the unit price",
     )
-    blk.adsorbent_unit_cost_coeff = pyo.Var(
-        initialize=4.58342,
-        units=pyo.units.USD_2020 * pyo.units.kg**-1,
-        doc="GAC adsorbent exponential cost pre-exponential coefficient",
-    )
-    blk.adsorbent_unit_cost_exp_coeff = pyo.Var(
-        initialize=-1.25311e-5,
-        units=pyo.units.kg**-1,
-        doc="GAC adsorbent exponential cost parameter coefficient",
-    )
-    blk.pres_other_cost_coeff = pyo.Var(
-        initialize=16660.7,
-        units=pyo.units.USD_2020,
-        doc="GAC pressure other cost power law coefficient",
-    )
-    blk.pres_other_cost_exp = pyo.Var(
-        initialize=0.552207,
-        units=pyo.units.dimensionless,
-        doc="GAC pressure other cost power law exponent",
-    )
-    blk.grav_other_cost_coeff = pyo.Var(
-        initialize=38846.9,
-        units=pyo.units.USD_2020,
-        doc="GAC gravity other cost power law coefficient",
+
+    # ---------------------------------------------------------------------
+    # correlation parameter data
+
+    blk.contactor_type_list = pyo.Set(
+        dimen=1,
+        initialize=[type.value for type in list(ContactorType)],
     )
-    blk.grav_other_cost_exp = pyo.Var(
-        initialize=0.490571,
-        units=pyo.units.dimensionless,
-        doc="GAC gravity other cost power law exponent",
+
+    contactor_cost_coeff_data = {
+        "pres": {0: 10010.9, 1: 2204.95, 2: -15.9378, 3: 0.110592},
+        "grav": {0: 75131.3, 1: 735.550, 2: -1.01827, 3: 0.000000},
+    }
+    adsorbent_unit_cost_coeff_data = {0: 4.58342, 1: -1.25311e-5}
+    other_cost_param_data = {
+        "pres": {0: 16660.7, 1: 0.552207},
+        "grav": {0: 38846.9, 1: 0.490571},
+    }
+    energy_consumption_coeff_data = {
+        "pres": {0: 8.09926e-4, 1: 8.70577e-4, 2: 0},
+        "grav": {0: 0.123782, 1: 0.132403, 2: -1.41512e-5},
+    }
+
+    blk.contactor_cost_coeff = pyo.Var(
+        blk.contactor_type_list,
+        range(4),
+        initialize=_unpack_data_dict(contactor_cost_coeff_data),
+        units=pyo.units.dimensionless,  # USD_2020 embedded in equation
```

Comment:
The USD_2020 comment can go above to avoid any funky black formatting that may arise